src/Sushi.Mediakiwi.API/Controllers/AssetController.cs

```diff
@@ -0,0 +1,271 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Sushi.Mediakiwi.API.Filters;
+using Sushi.Mediakiwi.API.Transport;
+using Sushi.Mediakiwi.API.Transport.Requests;
+using Sushi.Mediakiwi.API.Transport.Responses;
+using Sushi.Mediakiwi.Data.Configuration;
+using Sushi.Mediakiwi.Persisters;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Sushi.Mediakiwi.API.Controllers
+{
+    [MediakiwiApiAuthorize]
+    [Route(Common.MK_CONTROLLERS_PREFIX + "asset")]
+    [ResponseCache(NoStore = true, Location = ResponseCacheLocation.None)]
+    public class AssetController : ControllerBase
+    {
+        private static string Azure_Image_Container
+        {
+            get
+            {
+                return WimServerConfiguration.Instance?.Azure_Image_Container;
+            }
+        }
+
+        private static string Azure_Cdn_Uri
+        {
+            get
+            {
+                return WimServerConfiguration.Instance?.Azure_Cdn_Uri;
+            }
+        }
+
+        private static BlobPersister GetPersistor
+        {
+            get { return new BlobPersister(); }
+        }
+
+        protected Data.IApplicationUser MediakiwiUser
+        {
+            get
+            {
+                if (HttpContext.Items.ContainsKey(Common.API_USER_CONTEXT))
+                {
+                    return HttpContext.Items[Common.API_USER_CONTEXT] as Data.IApplicationUser;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
+        private System.Drawing.Image CreateThumbnailImage(System.Drawing.Image input)
+        {
+            try
+            {
+                var imageHeight = input.Height;
+                var imageWidth = input.Width;
+
+                var maxThumbWidth = System.Math.Max(64, WimServerConfiguration.Instance.Thumbnails.CreateThumbnailWidth);
+                var maxThumbHeight = System.Math.Max(48, WimServerConfiguration.Instance.Thumbnails.CreateThumbnailHeight);
+
+                if (imageHeight > imageWidth)
+                {
+                    var factor = ((float)maxThumbHeight / (float)imageHeight);
+                    imageWidth = (int)(factor * imageWidth);
+                    imageHeight = maxThumbHeight;
+                }
+                else
+                {
+                    var factor = ((float)maxThumbWidth / (float)imageWidth);
+                    imageHeight = (int)(factor * imageHeight);
+                    imageWidth = maxThumbWidth;
```

Comment:
Let's say we have an image that is 110x100 pixels and the max thumbnail size is on default (64, 48).
Then the factor will be based on the width, so factor = 64 / 110 = 0,5818.
Now the thumbnail height becomes 0,5818 * 100 = 58 pixels, which is more than the maximum thumb height of 48.
Should this not be allowed?

Reply:
@bart-van-der-drift great find !
will update accordingly